applications/OptimizationApplication/python_scripts/controls/shape/vertex_morphing_shape_control.py

```diff
@@ -181,93 +54,66 @@ def __init__(self, name: str, model: Kratos.Model, parameters: Kratos.Parameters
-            if len(fixed_model_parts_names)>0:
-                self.fixed_model_part_operation = ModelPartOperation(self.model, ModelPartOperation.OperationType.UNION, f"control_{self.GetName()}", fixed_model_parts_names, False)
-                self.fixed_model_part = self.fixed_model_part_operation.GetModelPart()
-                self.filter = KratosOA.NodalExplicitFilter(self.filter_model_part, self.fixed_model_part, self.parameters["filter_settings"]["filter_function_type"].GetString(),
-                                                           self.parameters["filter_settings"]["damping_function_type"].GetString(),
-                                                           self.parameters["filter_settings"]["max_nodes_in_filter_radius"].GetInt())
-            else:
-                self.filter = KratosOA.NodalExplicitFilter(self.filter_model_part, self.parameters["filter_settings"]["filter_function_type"].GetString(),
-                                                           self.parameters["filter_settings"]["max_nodes_in_filter_radius"].GetInt())
-
-            filter_radius_field = Kratos.Expression.NodalExpression(self.filter_model_part)
-            Kratos.Expression.LiteralExpressionIO.SetData(filter_radius_field, self.parameters["filter_settings"]["radius"].GetDouble())
-            self.filter.SetFilterRadius(filter_radius_field)
-            self.control_field = Kratos.Expression.NodalExpression(self.filter_model_part)
-            Kratos.Expression.LiteralExpressionIO.SetData(self.control_field,[0,0,0])
-        else:
-            self.filter.Initialize()
-            self._SetFixedModelPartValues(False)
-            self.control_field = self.filter.UnFilterField(self.GetPhysicalField())
-            self._SetFixedModelPartValues(True)
+        self.filter.SetComponentDataView(ComponentDataView(self, self.optimization_problem))
+        self.filter.Initialize()
+
+        # getting UnfilteredField of the PhysicalField is not possible with
+        # explicit solvers, and is not useful for this control hence, this
+        # will only update the shape changes to the existing shape. Hence,
+        # taking the actual physical field as the control field initially.
+        self.control_field = self.GetPhysicalField()
 
-        # initialize_mesh_motion
-        if self.has_mesh_motion:
+        # initialize_mesh_motion if defined.
+        if self.use_mesh_motion_solver:
             self._mesh_moving_analysis.Initialize()
+            # since this is applied on the boundary surface
+            # we can fix MESH_DISPLACEMENT on all skin model part nodes.
+            nodes_array = ModelPartOperation(self.model, ModelPartOperation.OperationType.UNION, f"control_{self.GetName()}_skin", self.parameters["skin_model_part_names"].GetStringArray(), False).GetModelPart().Nodes
+            Kratos.VariableUtils().ApplyFixity(Kratos.MESH_DISPLACEMENT_X, True, nodes_array)
```

Comment:
get an input to fix which directions, skin_model_part is not self explaining, I would suggest fixed model parts

Reply:
will be done :). Found a better modular way for this. I will put it.